src/wallet/rpcwallet.cpp

```diff
@@ -30,10 +31,27 @@
 
 #include <univalue.h>
 
+static const std::string WALLET_ENDPOINT_BASE = "/v1/wallet/";
+
 CWallet *GetWalletForJSONRPCRequest(const JSONRPCRequest& request)
 {
-    // TODO: Some way to access secondary wallets
-    return vpwallets.empty() ? nullptr : vpwallets[0];
+    LOCK(cs_main); // for now, protect vpwallets via cs_main
```

Comment:
Can we establish whether vpwallets is supposed to be protected by cs_main or not?